tfcoreml/_tf_coreml_converter.py

```diff
@@ -568,30 +603,56 @@ def convert(tf_model_path,
       dictionary containing the op's inputs that are constants and their values (as numpy arrays).
       The function can add custom layers or any other combination of CoreML layers to translate the TF op. 
       See "examples/custom_layer_examples.ipynb" jupyter-notebook for examples on using this argument. 
-
-  use_coreml_3: bool
-      Flag to switch to coremltools.converters.tensorflow which uses new Core ML 3 features.
+    
+  custom_shape_functions: dict(): {Text: func()}
+      Argument to provide user-defined functions to compute shape for given op.
+      A dictionary with keys corresponding to the type of TF Op and value as hadnled to user-defined function.
+      Function receives `layer specification` and `input shape` as a input.
+      output of the function must be output shape for give op. (generally List).
+      Custom shape function is required for adding custom layer in Core ML 3.
```

Comment:
Custom shape function is required for adding custom layer when `target_ios` is greater than '12'. 
Couple of things:
- raise an error right away if custom_conversion_fucntion is provided but the shape function has not been given
- add a test that uses both these arguments and point to that test as an example in this description or in the readme in general

Reply:
- `custom_conversion_function` does not require `custom_shape_function` as it's `custom_conversion_functions's` responsibility to update the shape. Will be pointing to the example for devs understanding
- Added test case with both `custom_conversion_function` and without
- Without `custom_conversion_function`, `custom_shape_function` is mandatory and we raise ValueError while adding custom layer if shape function is not provided
- Will be adding permanent link in example once this PR goes in